tracer/src/Datadog.Trace/Span.ISpan.cs

```diff
@@ -76,5 +76,8 @@ bool ISpan.Error
 
         /// <inheritdoc />
         string ISpan.GetTag(string key) => GetTag(key);
+
+        /// <inheritdoc />
+        void ISpan.AddEvent(SpanEvent spanEvent) => AddEvent(spanEvent);
```

Comment:
If we add the `AddEvent` method to the public span API, I actually think we will want to change the argument from the `SpanEvent` type to the primitive types it contains instead, so the arguments would be `string name`, `DateTimeOffset timestamp`, and `List<T> attributes`.

I don't want to prolong this PR since other work is waiting on this, so can we remove the public API additions and `Datadog.Trace.Manual` changes, and then re-address the public API in a separate PR?

Reply:
> so the arguments would be string name, DateTimeOffset timestamp, and List<T> attributes.
This seems reasonable!!

Reply:
Removed the public API and will revisit this when prioritized by product.